brain/circuit.cpp

```diff
@@ -103,10 +103,20 @@ class Circuit::Impl
     {}
     virtual ~Impl() {}
 
+    virtual size_t getNumNeurons() const = 0;
+
     GIDSet getGIDs( const std::string& target ) const
     {
-        return brion::Target::parse(
-            _targetParsers, target.empty() ? _circuitTarget : target );
+        if( target.empty( ))
+        {
+            GIDSet gids;
+            for( size_t i = 0; i < getNumNeurons(); ++i )
+                gids.insert( i + 1 );
+            return gids;
+        }
+        else
```

Comment:
OK, following our discussion, there are actually 3 cases: A target is specified and we use it, a target is not specified and we may either return all the GIDs or use the target that is specified in the config file. In the end, we decided to leave this getGIDs as it was, and let the calling application deal with what to do whenever no GIDs are returned. Still, the getNumNeurons method is needed for dealing with such cases.